src/worker/publisher/metrics_publisher.py

```diff
@@ -289,6 +292,11 @@ def get_tags_from_metric(self, metric_labels):
 
     # Publish metrics every 20 seconds
     while True:
-        raw_metrics = metricsPublisher.get_metrics()
-        metricsPublisher.publish_metrics(raw_metrics)
+        try:
+            raw_metrics = metricsPublisher.get_metrics()
+            metricsPublisher.publish_metrics(raw_metrics)
+        except URLError as e:
```

Comment:
Here is a lint issue, you may also print e in the log:
./src/worker/publisher/metrics_publisher.py:298:9: F841 local variable 'e' is assigned to but never used

Reply:
Thanks! I just fixed that. `logger.exception` will include the exception and stack trace, so it's not used...